Update twitter gem

colinsurprenant opened the pull request:
Update to twitter gem v6.2.0 which allow removing 2 obsolete monkey-patches.
This gem update removes the dependency on the faraday gem which has a version conflict per  elastic/logstash/pull/10934 
Integration tests delays have been also increased.

yaauie commented on lib/logstash/inputs/twitter.rb:
```diff
@@ -104,8 +105,6 @@ class LogStash::Inputs::Twitter < LogStash::Inputs::Base
   config :rate_limit_reset_in, :validate => :number, :default => 300
 
   def register
-    require "twitter"
```
IIRC the reason to require the library in `register` is to only "activate" the dependency when this plugin is registered, which reduces the footprint of a process that has a plugin installed but no pipeline using it.

Since `logstash-input-twitter` is _not_ bundled with any logstash distributions, I don't think this is terribly likely to occur, and since we are loading portions of the twitter library to monkey-patch them anyway, I am in favor of moving the whole require up as you have done.

colinsurprenant replied:
The purpose of this PR is to be able to move forward with https://github.com/elastic/logstash/pull/10934 where the twitter input will be included as a default plugin. 
But in any case, IIRC we also decided some time ago to move away from this idea of requiring dependant libraries in the `register` method in particular to fail early in the build process if there was any conflicts between dependant libraries.

yaauie commented on lib/logstash/inputs/twitter/patches.rb:
```diff
@@ -10,54 +8,12 @@ class TwitterPatches
-      end
-
-      def self.patch_connection_stream
-        ::Twitter::Streaming::Connection.class_eval do
```
Would you mind linking to an upstream issue/PR that indicates _why_ these monkey-patch is obsolete with twitter gem v6.2.0?

colinsurprenant replied:
I did a visual verification, first to identify the nature of the patch to start with by comparing with the original code in v5.15.0 and then looking at the code in 6.2.0 and concluded that these 2 patches were not required anymore. Also all tests are passing but TBH I don't know if there are specific tests for these 2 patches purposes.

colinsurprenant commented:
Turns out that `public_suffix` 3.x+ requires ruby syntax 2.1+ so logstash logstash >= 6.x is required.
I added this in the gemspec and  removed logstash v5 from the builds.
```
s.required_ruby_version = '>= 2.1.0'
```

colinsurprenant commented:
@robbavey @yaauie when you can let me know if you are good with the last changes & comment responses.

colinsurprenant commented:
Merged and published v4.0.1 - Thanks @robbavey @yaauie !
